src/accel/SharedParams.cc

```diff
@@ -430,25 +547,44 @@ void SharedParams::initialize_core(SetupOptions const& options)
  */
 void SharedParams::try_output() const
 {
-    CELER_EXPECT(params_);
-    if (output_filename_.empty())
+    if (!output_reg_)
     {
+        // No output registry exists (either independently of setting up
+        // Celeritas or when calling Initialize)
         return;
     }
 
-#if CELERITAS_USE_JSON
-    CELER_LOG(info) << "Writing Celeritas output to \"" << output_filename_
-                    << '"';
-
-    std::ofstream outf(output_filename_);
-    CELER_VALIDATE(
-        outf, << "failed to open output file at \"" << output_filename_ << '"');
-    params_->output_reg()->output(&outf);
-#else
-    CELER_LOG(warning)
-        << "JSON support is not enabled, so no output will be written to \""
-        << output_filename_ << '"';
-#endif
+    std::string filename = output_filename_;
+    if (CELERITAS_USE_JSON && !params_ && filename.empty())
+    {
+        // Setup was not called but JSON is available: make a default filename
+        filename = "celeritas.json";
+        CELER_LOG(debug) << "Setting default Celeritas output filename";
+    }
```

Comment:
Since `output_filename_` is set in `initialize_core()`, if offloading is disabled this default name will be used instead of the filename in the `SetupOptions`.

Reply:
I think that's the only generic way to do this... we always load this through `celer-g4` but user apps won't usually provide a `SetupOptions`. Any ideas how to make both work better?

Reply:
Brilliant! Added.